### ISSUE TITLE
Merging code bases

### ISSUE THREAD

**Issue by gongled** [cscart/server-ansible-playbooks#3]
Hi there.

I suggest to merge [gongled/cartomatic](https://github.com/gongled/cartomatic) into this project's code base. It will be useful for everyone who develops for CS-Cart and Multi-Vendor platforms. I've already released new version that supports the following features:

- HTTP/2 support
- Performance tuning for highload
- Deflate/GZIP compression
- Google Pagespeed filters support
- Well-looking error pages
- SSL/TLS encryption via Let's Encrypt (optional)
- Strong SSL preferencies (A+ by default on SSL Labs)
- Development environment based on Vagrant
- Configurations for creating your own images based on Packer

Unfortunately, it doesn't work for Ubuntu/Debian, but I have [my own repository](https://github.com/gongled/gongled-repo) with bunch of packages which works perfectly for CentOS/RHEL 6/7.

**Comment by alpharder** [comment]
Hi!

We're currently using Ubuntu Server in production. So, it would be great if your changes also work for Ubuntu.

**Comment by gongled** [comment]
I'm totally agree with you, but It's not easy to do. Unfortunately, I don't have any public repositories with packages for Ubuntu/Debian. That's why we have only two options:

* Implement the difficult logic via Ansible for big amount of platforms as it was before. It was hard to maintain and I abandoned this approach in favor of simplicity. Now it works faster and efficiently.
* Waste a lot of time to create repository for these platforms. It's hard to maintain if you have so many distros (from Ubuntu 14.04 to Ubuntu 16.04, from Debian Squeeze to Debian Jessie) instead of CentOS/RHEL 6 and 7 versions only.

Besides, I know that more than half of your servers works on CentOS/RHEL. Also I think that you don't have any platform-specific software which you can use only on Ubuntu/Debian. I'm pretty sure that after a few weeks of adaptation your employees won't see any differences between CentOS/RHEL and Debian-based systems.

**Comment by alpharder** [comment]
@gongled Your point is clear to me. Also, the features you listed above are greatly needed.

But I have to mention the following: RHEL-based distros are great for production environment, but Ubuntu Server requires much lower technical level to handle common tasks. The internet is full of articles, tutorials and howtos related to Ubuntu Server configuring and management. 
I don't talk about Debian directly as well as supporting 14.04 to 16.04. In my opinion it is only worth being focused on LTS releases.

Dropping Ubuntu Server support comes with dropping a large audience of entry-to-mid level developers that want to check out things right here - this is what I worry about.

**Comment by alpharder** [comment]
The Cartomatic itself is a great and solid product built by a professional server engineer. We're highly interested in merging our codebases at least in the separate branch.

**Comment by gongled** [comment]
Good news everyone. 

Me and @imakarov discussed and we made decision to merge my project with backward compatibility with Ubuntu/Debian. I'm going to implement wrapper which allows to use legacy version (0.1) without any changes of code for Ubuntu/Debian and CentOS/RHEL with new modern version (0.2).

Although I'm going to support Ubuntu/Debian, I focused on RHEL-based systems. I hope that I'll find the strength to prepare repository for Ubuntu/Debian OS to be able to deploy software for all these platforms identically. 

Anyway, I'm glad to say that now we're working together.
